### PR TITLE
[ci] Temporary mitigation for the flaky tests on the CW310

### PR DESCRIPTION
**Pull request by milesdai**
This PR adds two temporary mitigations to address the communications failures being observed on the CW310 in CI.
- Adding `--notest_keep_going` to the cw310 tests prevents tests from queueing up on the FPGA runner when opentitantool's uart connection times out
- Reducing the number of CW310 tests being run in CI decreases the frequency of the FPGA failures.

This is a temporary measure as we diagnose the issue. As tracked in #13044, these changes should be undone after this is fixed.

### CODE REVIEW

**Review by timothytrippel — APPROVED** [review]

**Review by drewmacrae — APPROVED** [review]
I'll try to keep an eye on the other tests when they roll up the easy to test bitstreams

File: azure-pipelines.yml [hunk 1]
```diff
@@ -406,24 +406,7 @@ jobs:
...
-      export BITSTREAM="--offline --list ${SHA}"
-      ci/bazelisk.sh test \
-          --define DISABLE_VERILATOR_BUILD=true \
-          --test_tag_filters=cw310,-broken \
-          --test_output=all \
-          --build_tests_only \
-          --define cw310=lowrisc \
-          $(./bazelisk.sh query 'rdeps(//...,@bitstreams//:bitstream_test_rom)') \
-          # All the tests that depend on the test rom
+      ci/scripts/run-fpga-cw310-tests.sh
```
Comment (drewmacrae) [inline azure-pipelines.yml:409]: Nit can you file a follow-on issue: moving this to a script isn't enough to make it portable but implies it is. This script wont work without the bitstream and isn't testing for it's presence.

File: ci/scripts/run-fpga-cw310-tests.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+echo "" >> ${BITCACHE_DIR}/lowrisc_systems_chip_earlgrey_cw310_0.1.bit.splice
+echo -n ${SHA} > ${BITCACHE_DIR}/../../latest.txt
+export BITSTREAM="--offline --list ${SHA}"
+
+# TODO: remove the --notest_keep_going and --nokeep_going flag after the CW310
+# tests are reliable (#13044)
+ci/bazelisk.sh test \
+    --define DISABLE_VERILATOR_BUILD=true \
+    --notest_keep_going \
+    --nokeep_going \
```
Comment (drewmacrae) [inline ci/scripts/run-fpga-cw310-tests.sh:24]: More Nits, just things you should know they don't need fixing here: I think this is the default behavior for builds, it's fine to have it.

File: ci/scripts/run-fpga-cw310-tests.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+BITCACHE_FILE=$BITCACHE_DIR/lowrisc_systems_chip_earlgrey_cw310_0.1.bit.orig
+cp $BIN_DIR/hw/top_earlgrey/lowrisc_systems_chip_earlgrey_cw310_0.1.bit ${BITCACHE_FILE}
+echo "" >> ${BITCACHE_DIR}/lowrisc_systems_chip_earlgrey_cw310_0.1.bit.splice
+echo -n ${SHA} > ${BITCACHE_DIR}/../../latest.txt
+export BITSTREAM="--offline --list ${SHA}"
+
+# TODO: remove the --notest_keep_going and --nokeep_going flag after the CW310
+# tests are reliable (#13044)
+ci/bazelisk.sh test \
+    --define DISABLE_VERILATOR_BUILD=true \
```
Comment (drewmacrae) [inline ci/scripts/run-fpga-cw310-tests.sh:22]: More Nits, just things you should know they don't need fixing here: This isn't necessary, this shouldn't include the verilator tests anymore

File: azure-pipelines.yml [hunk 4]
```diff
@@ -406,24 +406,7 @@ jobs:
         - sw_build
   - bash: |
       set -e
```
Comment (drewmacrae) [inline azure-pipelines.yml:408]: More Nits, just things you should know they don't need fixing here:  I think a 1 line script doesn't need set -e to return the return value of the solitary line.

File: azure-pipelines.yml (hunk 1 above)
Comment (drewmacrae) [reply azure-pipelines.yml:409]: Moving it to a script is likely to become permanent because it's useful.

**Review by drewmacrae — APPROVED** [review]
None of these nits are critical, They're just things we should document.

File: azure-pipelines.yml (hunk 1 above)
Comment (milesdai) [reply azure-pipelines.yml:409]: Yeah I think I will need to some more work to figure out where it's searching for the bitstream.

File: ci/scripts/run-fpga-cw310-tests.sh (hunk 2 above)
Comment (milesdai) [reply ci/scripts/run-fpga-cw310-tests.sh:24]: It is the default, but we're modifying this in `ci/.bazelrc`, so I'm undoing it again here just for this fix.

File: azure-pipelines.yml (hunk 1 above)
Comment (milesdai) [reply azure-pipelines.yml:409]: This is now tracked in #13070

**Pull request merged**